### ISSUE TITLE
 Standard names: *ocean model grid information*

### ISSUE THREAD

**Issue by jmecki** [cf-convention/vocabularies#219]
Before submitting an issue be sure you have read and understood the [rules for vocabulary changes](http://cfconventions.org/standard_name_rules.html) and review the [guidance for constructing standard names](http://cfconventions.org/Data/cf-standard-names/docs/guidelines.html)

Please note that it is fine to group together a number of proposals in a single GitHub issue (i.e. it is not necessary to open a separate issue for each vocabulary term). Change proposals should include the following information as applicable.

**Proposer's name** 
Jenny Mecking

**Date**
Sept. 8 2024

For each term please try to give the following:

**- Term** 
eastward_ocean_gridbox_length_at _t_point

**- Description**
The length of the gridbox in the east/west direction for the t-point gridbox 

**- Units** (If applicable).
m

**- Term** 
northward_ocean_gridbox_length_at _t_point

**- Description**
The length of the gridbox in the north/south direction for the t-point gridbox (i.e dy on t-points)

**- Units** (If applicable).
m

**- Term** 
eastward_ocean_gridbox_length_at _u_point

**- Description**
The length of the gridbox in the east/west direction for the u-point gridbox 

**- Units** (If applicable).
m

**- Term** 
northward_ocean_gridbox_length_at _u_point

**- Description**
The length of the gridbox in the north/south direction for the t-point gridbox (i.e dy on u-points)

**- Units** (If applicable).
m

**- Term** 
eastward_ocean_gridbox_length_at _v_point

**- Description**
The length of the gridbox in the east/west direction for the v-point gridbox 

**- Units** (If applicable).
m

**- Term** 
northward_ocean_gridbox_length_at _v_point

**- Description**
The length of the gridbox in the north/south direction for the t-point gridbox (i.e dy on v-points)

**- Units** (If applicable).
m

**Comment by JonathanGregory** [comment]
Dear Jenny @jmecki

It is interesting that standard names haven't previously been defined for the horizontal size of gridcells. Usually this information is not necessary to store as a data variable, because you can calculate it when needed from the bounds of the latitude and longitude coordinate variables. Please could you describe why these standard names are needed?

Best wishes and thanks

Jonathan

**Comment by ChrisBarker-NOAA** [comment]
I'd note also that these would only apply for north-east aligned rectangular grids. 

*maybe* there's a more general standard name that could be used?

**Comment by jmecki** [comment]
Thank you for your responses,  it's the first time I've proposed variable names so I might have missed something when I looked things up.

In response to Jonathan's comment, while you can compute it from the latitude and longitude (bounds/vertices) models often artificially narrow channels, for example the Strait of Gibraltar and in the Indonesian Throughflow region.  This is especially true for models that have ocean model resolution similar to the models used in CMIP6 (i.e. 1 degree), therefore giving incorrect values.

In response to Chris Barker's comment, what I had in mind were the lengths/widths of the grid boxes which if they are on an irregular grid then they wouldn't be strictly northward or eastward but along the model grid lines.  

Maybe the names:
ocean_gridbox_length_at _t_point
ocean_gridbox_width_at _t_point
ocean_gridbox_length_at _u_point
ocean_gridbox_width_at _u_point
ocean_gridbox_length_at _v_point
ocean_gridbox_width_at _v_point

However, how length and width are defined might be unclear.

**Comment by ChrisBarker-NOAA** [comment]
> models often artificially narrow channels

very interesting -- I had no idea. So yes, this does make sense to me as something to capture in a standard_name

Is "gridbox" the correct term, however? though I have idea what a bette tem might be:

channel_width? or some such?

> However, how length and width are defined might be unclear.

yes, that's a trick -- "x" and "y" -- I can't recall if those terms are currently used to mean "logical x [y]" rather than literal.

**Comment by atreguier** [comment]
The convention so far in standard names is to use x and y, e.g., "ocean_heat_x_transport: "x" indicates a vector component along the grid x-axis, positive with increasing x"

**Comment by ChrisBarker-NOAA** [comment]
thanks @atreguier -- then "x" and "y" do seem to mean the correct thing in this context.

**Comment by martinjuckes** [comment]
Could this information be provided using the [mesh topology attributes](https://cfconventions.org/cf-conventions/cf-conventions.html#appendix-mesh-topology-attributes) defined in Appendix K of the CF conventions?

**Comment by JonathanGregory** [comment]
Dear Jenny

I'm a bit puzzled. In answer to my question above you agreed you can get gridcell dimensions from the cell bounds, but you say that these are not necessarily the "right" answers. Surely they are the true answers for the _grid_, even if not for _reality_. Is it perhaps not the gridcell dimensions that you want to record and give standard names to, but the real distances across straits etc?

Best wishes

Jonathan

**Comment by taylor13** [comment]
Hi Jenny,

I would find it helpful if you could describe how a model actually treats the processes in these "special" grid cells where somehow the equations governing the ocean are modified to take into account a channel that is narrower than the what the cell can represent.  Is there a reference for how they are treated?

thanks,
Karl

**Comment by atreguier** [comment]
Dear Karl, dear Jenny, 
I know that adjusting the width of a one-grid-point-wide strait (narrowing it to reduce the volume flux) was done often in the NEMO model. It may be referenced in the manual, I am not sure. Even when no "ad hoc" modification has been made, in NEMO the "scale factors" (the dx,dy, used in the disctretized equations) are not exactly the distances between grid points (to machine accuracy) because they are computed analytically from the equations describing the grid. So, in order to close the mass budgets, we need to use the original "scale factors" of the model. I do not know how it is for other models, especially fine volume models with triangular meshes. It would certainly be interesting for modelling groups to publish on esgf everything a user needs to compute budgets.

**Comment by jmecki** [comment]
Thank you Anne-Marie,  I agree with what you are saying and I am also most familiar with NEMO.  The variables I'm looking for are basically the equivalent of the NEMO variables 'e1t, e2t, e1u, e2u, etc...'  while the e3* variables are a bit easier to estimate (I also opened the issue 220 for these).  Not having this specifically has lead to difficulties in estimating transports in models.  I think just having widths of channels will not be sufficient because the channels with artificial narrowing might end up being different in different models and resolutions. I tried estimating the grid box dx/e1* and dy/e2* (lengths and widths) from the information provided in the CMIP5 and CMIP6 database at T,U and V points and have found that several models have had this issue but mainly the NEMO based ones.

**Comment by jmecki** [comment]
Has there been a decision made about having a name for this?  It would be useful to have one even if all models might not be able to provide this field.

Would a possible suggestion now be:

ocean_gridbox_x_length_at _t_point
ocean_gridbox_y_length_at _t_point
ocean_gridbox_x_length_at _u_point
ocean_gridbox_y_length_at _u_point
ocean_gridbox_x_length_at _v_point
ocean_gridbox_y_length_at _v_point

**Comment by JonathanGregory** [comment]
Dear Jenny

Speaking for myself, I don't think it's yet clear what these quantities actually are. Please could you describe them in geophysical terms, as we need for standard names? How do you use them in a calculation, for example? That might clarify the issue. I think that "gridbox lengths" would generally be understood as the distance between coordinates, as I interpreted it before.

Also, I don't think we would mention T, U and V points in standard names. CF doesn't have a notion of grid arrangements, to which that refers. (Maybe it should, as has been discussed a few times, but that's not the issue here.) I think we need to give a name to a distance needed for some purpose; the gridpoints it applies to should be obvious from the coordinates or bounds.

Please don't give up. These discussions are often difficult, but usually achieve a good result!

Best wishes

Jonathan

**Comment by fmassonn** [comment]
Hi,

I agree that having these terms would be great to recompute fluxes, or to compute fluxes over non-standard straits.

François

**Comment by jmecki** [comment]
Hi Jonathan and Taylor,

I hope this helps answer some of the questions.

I would use these variables as follows:

To compute the volume transport along a constant line in the y/j direction:

$volTrans(t)=\int^{xmax}_{xmin}$ $\int^{surface}_{H}$

**Comment by ChrisBarker-NOAA** [comment]
I edited the TeX in the post -- I *think* I got it right, but please do correct it if I misunderstood what it was supposed to be.

But as to the topic -- in some models (ROMS), what is computed is the flux through a gird cell -- though it may be provided in the output as a velocity -- in that case, you'd need to know the cross sectional area of the channel -- and it that doesn't match the area of the cell face, you'd need to know that -- is that what this is for?

**Comment by jmecki** [comment]
> I edited the TeX in the post -- I _think_ I got it right, but please do correct it if I misunderstood what it was supposed to be.
> 
> But as to the topic -- in some models (ROMS), what is computed is the flux through a gird cell -- though it may be provided in the output as a velocity -- in that case, you'd need to know the cross sectional area of the channel -- and it that doesn't match the area of the cell face, you'd need to know that -- is that what this is for?

Thanks, I was really struggling with the latex in github and accidentally posted it before it was finished.  I will post a more complete response once I have it typed up better.

**Comment by jmecki** [comment]
Hi All,

Sorry it took me a while to respond.

A simple example for which I would use dx on the v grid points would be to compute the volume transport through a section as follows:

![Screenshot 2024-11-08 at 16 04 40](https://github.com/user-attachments/assets/9dd54a2a-1811-465e-a61d-81d1d6d27b0f)

where you would have to compute dx on the v points either from the vertices or the centre points of the grid points of the grid boxes where the computation would differ depending on the Arakara grid used in the model.  While this isn't too complex it gets more complex when changing to temperature/heat and salinity/freshwater transports where you have to move all the data (temperature, salinity and velocity) to the same grid point, typically along the grid box edge of the t-points.    This again is dependent on the model grid used and having dx, dy and dz on the different grid points (i.e. t,u,v,etc).

In terms of the comment related to ROMS, yes this is something that it would be used for.

Furthermore, as mentioned before if the grid boxes have an artificially narrowed point for some channels, often estimating dx, dy, and dz give the wrong value leading to spikes that shouldn't be there...

In Mecking and Drijfhout 2023 in the methods we made the comment:  'In global and Indo-Pacific computations of OHT there are spikes at the latitudes which are impacted by Indonesian throughflow in some models. Several ocean models which do not have high enough horizontal resolution to resolve the narrow ocean channels, like the ones present in the Indonesian throughflow, artificially narrow these channels but only on either the U- or V-grid points. The information available in the CMIP5/6 archives only contains information about the grid size on the T-grid points. For the models where these spikes occur, we removed the data from the latitudes where the spikes appear.'

While it is possible to compute dx and dy from the grid box vertices this is not always provided and it adds extra computations and may not give the correct values, especially if there are artificially narrows channels.  Please correct me if I'm wrong, I'm not a model developer, but from the ocean model code I have looked at, in both Arakawa-B and Arakawa-C grid models they used dx and dy values at the t,u and v grid points in the computations as opposed to the vertices.  I believe that it would still be very useful to have names for these variables even if there might be work arounds.

**Comment by ChrisBarker-NOAA** [comment]
> Furthermore, as mentioned before if the grid boxes have an artificially narrowed point for some channels, often estimating dx, dy, and dz give the wrong value leading to spikes that shouldn't be there...

> While it is possible to compute dx and dy from the grid box vertices this is not always provided and it adds extra computations and may not give the correct values, especially if there are artificially narrows channels. 

I think that "artificially narrows channels" is the key point here -- otherwise, the grid is well defined, and I don't think we need standard names. But if there's been an adjustment in the model, such that you can't compute the correct flux from the grid geometry, then you absolutely need to know the "virtual" channel size.

So I support the idea -- but have no opinion about how to exactly spell the name :-)

**Comment by atreguier** [comment]
Hi everybody, I just realized that this procedure to artificially narrow some straits  is well explained in the NEMO manual , with a nice figure. Here it is 
[nemo_manual_handmade_grid_corrections.pdf](https://github.com/user-attachments/files/17686527/nemo_manual_handmade_grid_corrections.pdf)

**Comment by JonathanGregory** [comment]
Dear Jenni @jmecki et al.

Thanks for explaining. I agree that working out _dx_ and _dy_ on velocity points from the tracer grid is quite intricate, so it's convenient to record them as data variables with metadata to identify them. We can regard _dx_ and _dy_ as metrics of the grid. We already have a standard name for _dz_, namely `cell_thickness`. It would be consistent to use standard names of `cell_x_width` and `cell_y_width` for _dx_ and _dy_ respectively. The coordinates of the variable containing the quantity show what grid it's on, so that shouldn't be in the standard name. To make the correct variables easier to find, we could define `thickness`, `x_width` and `y_width` as keywords for the `cell_measures` attribute, which currently supports only `area` and `volume`.

However, if straits have effective widths that differ from what the gridpoint coordinates indicate, that's not a metric of the grid, and you can't work them out. I don't exactly understand your equation, because of three appearances of _v_ on the right-hand side. I think we can write the northward volume transport in `m3 s-1` through a section at latitude row $j$ as

$$V(j,t) = \sum_{i=\mbox{west}}^{\mbox{east}} \sum_{k=\mbox{surface}}^{\mbox{bottom}} \delta V(i,j,k,t)$$

where

$$\delta V(i,j,k,t) = v(i,j,k,t)\ dx'\ dz$$

is the contribution from gridbox $(i,j,k)$ to the transport, where $dx'$ is the effective width of the cell for transport and $dx'\leq dx$, the cell x-width. Is that right? If so,

$$dx' = \frac{\delta V/dz}{v}$$

This is the ratio of the `ocean_volume_y_transport` $\delta V$ (an existing standard name, unit `m3 s-1`) per unit thickness $1/dz$ to `sea_water_y_velocity` $v$ (an existing standard name, unit `m s-1`). Hence we could give $dx'$ the standard name `ratio_of_ocean_y_transport_per_unit_thickness_to_sea_water_y_velocity`. Would that make sense to you?

Best wishes

Jonathan

PS How marvellous that you can write $\TeX$ in markdown. I didn't know that before!

**Comment by jmecki** [comment]
Hi All,  sorry about all the v's in the equation.   That's my bad, in my head I think of dx on the v grid points as dxv and similarly dy as dyv and dz as dzv I should have been more precise and defined this.  I like Jonathan's suggestion about using 
cell_x_width and cell_y_width, that makes it clear for me.

**Comment by JonathanGregory** [comment]
Dear Jenni

I think `cell_x_width` and `cell_y_width` would be OK for the widths that you could calculate from gridpoints (from the tracer gridpoints for the cells of the velocity grid, for instance). I fear it might be misleading to use these names for the quantities which are "artificially" narrowed, using @atreguier's word. That's why I suggest the longer term `ratio_of_ocean_y_transport_per_unit_thickness_to_sea_water_y_velocity`, or the analogous one for `x`, which I believe describes the physical meaning of the quantity you're interested in. What do you and others think of that?

Best wishes

Jonathan

**Comment by jmecki** [comment]
Hi Jonathan,

While I believe your suggestion would be technically more accurate, I think it might be confusing and misleading to people trying to find the variable they are looking for (i.e. dx and dy for grid boxes of interest).  That name also suggests that it might be time variable, which shouldn't be the case.  Therefore, I prefer the simpler names for it i.e. cell_x_width and call_y_width.  Is there a possibility to include in the description of the variable name that they should include manually/artificially narrowed grid box widths?

**Comment by JonathanGregory** [comment]
It's possible someone might want a standard name for the true width of the cell, which would be inconvenient if we had defined `cell_x_width` to mean an artificially narrowed version which _isn't_ really the cell width. How about `cell_x_width_for_ocean_transport`?

**Comment by taylor13** [comment]
I agree that if we go with "cell width", it needs to be modified in some way.  "for ocean transport" is good.  I had thought of "modified_cell_x_width", which is more general, but perhaps less easily understood.

**Comment by ChrisBarker-NOAA** [comment]
>  in NEMO the "scale factors" (the dx,dy, used in the disctretized equations)

>  The variables I'm looking for are basically the equivalent of the NEMO variables 'e1t, e2t, e1u, e2u, etc...' 

what does NEMO call these in its docs? are there "wordier" terms than `e1`, `e2` ?

I really don't like this: `ratio_of_ocean_y_transport_per_unit_thickness_to_sea_water_y_velocity` -- yes, that's how you can use it, but it's kind of like using `ration_of_force_to_acceleration` rather than `mass` :-)

`modified_cell_x_width` is getting closer -- but not ideal -- `virtual_cell_width`, `analytical _cell_width`, `computational_cell_width` `effective_cell_width`? -- spitballing here ....

**Comment by JonathanGregory** [comment]
@ChrisBarker-NOAA. Are you happy with `cell_x_width_for_ocean_transport`? That describes its purpose, I think.

**Comment by JonathanGregory** [comment]
Dear Jenny @jmecki 

I agree with Karl, because standard names are intended to identify well-defined geophysical quantities, so that data-users can tell whether variables from different sources are the "same thing". "Effective cell width" is generic. In other models or applications, other quantities might be described like that. That is why I suggested "for transport", as a description of the purpose for which this version of the cell width is used. Karl's question is useful: what other kinds of calculation use this effective cell width? If my suggestion is too specific, we can probably think of something more general that describes the physical purposes for this quantity.

I think the dx and dy values computed from the gridpoints would be called `cell_x_width` and `cell_y_width` if you need standard names for those too. As you've explained, the effective width for transport calculations is a "modified" cell width, not the _actual_ cell width, so they shouldn't share the same standard name. They have different geoscientific purposes.

No, we would not indicate in the standard name which of the grids it applies to, because the grid is described by the coordinates. For different grids they could be given different PCDMI names (I don't know the practice with that), but they'd have the same CF standard name, because a physical quantity is the same regardless of the locations of its gridpoints.

Best wishes and thanks for pursuing this

Jonathan

**Comment by JonathanGregory** [comment]
Yes, I agree that `length` would be better than `width`. Thanks, David.

**Comment by ChrisBarker-NOAA** [comment]
>  standard names are intended to identify well-defined geophysical quantities, so that data-users can tell whether variables from different sources are the "same thing".

This is critical, actually -- it may well be that gird metrics are by definition model specific -- ROMS may use it differently than NEMO, etc. So perhaps a standard name is simply not appropriate for this kind of thing.

"Effective cell width" is generic. In other models or applications, other quantities might be described like that. That is why I suggested "for transport"

The generic name is deliberate, because the width (or length) or the cell is used for multiple calculations -- and outside of standard names anyway, we avoid having two ways to describe the same thing in CF -- i.e. generic is what we are going for :-)

But again, this may be pointing to this not being appropriate for a standard name -- are there any other analogous standard names for things like grid metrics?

@jmecki: I suggest you take a look at:

https://sgrid.github.io/sgrid/

It isn't (and may never be) a part of CF, but it is supposed to be CF compatible, and that may be a better place to define these sorts of metrics.

It's managed here:

https://github.com/sgrid/sgrid

And has not seen any activity for a while, but it is being used, at least a bit :-)

**Comment by JonathanGregory** [comment]
@ChrisBarker-NOAA. Yes, "we avoid having two ways to describe the same thing in CF". That applies to standard names as well. We take great care to avoid giving a new standard name to a quantity that already has one. But the cell length according to the grid is not the same thing as the "artificially narrowed" cell length for ocean transport calculations (or some other description that better describes what it is used for). That is why they need different standard names. Yes, it is consistent with past practice to give standard names to grid metrics. There are existing standard names of `cell_area` and `cell_thickness`.

**Comment by taylor13** [comment]
I think we need to wait to hear from @jmecki as to how the cell length quantity is used (besides in computing transports) before we can come up with a more general term that is still specific enough to be able to distinguish it meaningfully from simply "cell_length".

**Comment by ChrisBarker-NOAA** [comment]
The question is whether the cell metric at hand is used in all (many, most?) models the same way -- or at least having the same meaning. If so, then a standard name may make sense, if not, then maybe not.

**Comment by atb299** [comment]
Hi all,

As a suggestion, how about defining names for `cell_x_width` and `cell_y_width` for the widths that you could calculate from gridpoints, plus `cell_x_width_scale_factor` and `cell_y_width_scale_factor`? This could be a clearer and more generic way for modelling centres to provide the information that is required. The names could be used for any model (ocean, atmosphere, ice, ...). There may be cases, existing or future, where cell widths are not the distance between adjacent grid points and/or are for use other than computing ocean transport.

**Comment by JonathanGregory** [comment]
Thanks for your comment, Adam @atb299. If it's easier to provide a scale factor than the modified cell length, we could certainly consider giving it a standard name. However, I still think it's necessary for the standard name to mention what the factor is intended for. The purpose of standard names is to indicate what quantities are "the same thing". Two quantities, both named "cell length scale factor", which come e.g. one from an ocean model, the other from a land ice model, are unlikely to refer to the geoscientific concept, so they shouldn't have the same standard name. If "ocean transport" is too limited to describe all the purposes for the scaled "artificially narrowed" cell length in NEMO, how would you characterise it, generally?

**Comment by taylor13** [comment]
I also would like to know what more general application is envisioned for this quantity beyond being needed to compute ocean transports.  This is why I made the [comment](https://github.com/cf-convention/vocabularies/issues/219#issuecomment-2477900013) above and the [follow up](https://github.com/cf-convention/vocabularies/issues/219#issuecomment-2479732923).

**Comment by atb299** [comment]
@JonathanGregory, @taylor13, I've spent a little time today trying to find out whether there are use cases beyond ocean transports, such as reducing volume or area of a cell which could affect calculation of volume integrals or surface fluxes for example. I've not found any yet, but that doesn't mean they don't exist.  

I have found that it's not just NEMO that this affects. Edited scale factors are mentioned in [NorESM](https://noresm-docs.readthedocs.io/en/noresm2/model-description/ocn_model.html#:~:text=The%20metric%20scale%20factors%20are%20edited%20to%20the%20realistic%20width%20of%20the%20Strait%20of%20Gibraltar%20so%20that%20strong%20velocity%20shears%20can%20be%20formed%2C%20enabling%20realistic%20mixing%20of%20Mediterranean%20water%20entering%20the%20Atlantic%20Ocean.).

**Comment by ChrisBarker-NOAA** [comment]
> I have found that it's not just NEMO that this affects.

thanks -- and I"m sure there are other model's that do similar things as well. The question is whether all (most, many?) models do these kinds of things the same way -- that is, is there a single "thing" that we can define a standard name for -- or is the meaning of these these grid metric adjustments particular to each model?

I've done a lot of work with ROMS results for example -- and I go to the ROMS docs to make sure I understand what the outputs mean. *maybe* they are the same for all Arakawa C grids -- or all grids? but I doubt it :-(

I guess what I'm getting at is that it may not be possible to have a. standard name for everything -- and that's OK.

**Comment by taylor13** [comment]
Nice find, @atb299!  There is one sentence in the NorESM documentation of real relevance:  

> The metric scale factors are edited to the realistic width of the Strait of Gibraltar so that strong velocity shears can be formed, enabling realistic mixing of Mediterranean water entering the Atlantic Ocean.

So apparently, one internal model use of the scale factors is to compute more realistic velocities (and shears), so that the mixing parameterization yields a more realistic result.  I think this still falls under the category of "for ocean transport" (if velocity can be thought of as a "transport of non-dimensional 'stuff' of unit value").

**Comment by larsbarring** [comment]
Here is a couple of thoughts from an outside perspective. My understanding is that this scale factor is used to mimic a more realistic channel geometry (in particular its cross-section) than what is allowed by a particular model grid resolution. This is done to allow the model to produce more realistic through-channel flow. As such this factor is not directly adjusting the flow as such.  

How about `grid_adjustment_factor_for_channel_flow`, if there is a x- and y-component of this factor one may add `_in_x_direction_` etc. Other phrases that comes to mind are `_scale_factor_`, `_cross_channel_geometry`, `_channel_cross_section`, `_channel_width` (noting that "width" is a length/distance measure more or less orthogonal to `channel_length`).

**Comment by atb299** [comment]
How was this issue resolved when the name "`thkcello`" was accepted? Models that use partial bottom cells define cell thicknesses that are different than the reference grid. Users have to rely on the `long_name` or other attributes to know which point on a grid cell the thickness refers to. Could use cases for `cell_x_width` be specified in the `long_name` attribute? This might give some flexibility to how it is used.

**Comment by atb299** [comment]
@larsbarring 
> As such this factor is not directly adjusting the flow as such.

I think I disagree with this statement. The cell widths are used by the model to compute transports/fluxes through the face of the model cell. If you try to use a different cell width (e.g. the distance between two f points on an Arakawa C grid) to compute transports using velocities offline you will get the wrong answer. Unless the cell widths used by the model are provided it is (almost) impossible to accurately calculate transports for these cells offline.

**Comment by JonathanGregory** [comment]
Adam @atb299 [wrote](https://github.com/cf-convention/vocabularies/issues/219#issuecomment-2485227157)

> How was this issue resolved when the name "`thkcello`/[`cell_thickness`](https://cfconventions.org/Data/cf-standard-names/current/build/cf-standard-name-table.html#:~:text=cell_thickness-,m,-change_in_atmosphere_energy_content_due_to_change_in_sigma_coordinate_wrt_surface_pressure)" was accepted? Models that use partial bottom cells define cell thicknesses that are different than the reference grid.

That's a good question. I don't remember, but I suspect it wasn't discussed. The `cell_thickness` is probably used for doing vertical integrals, for which it should be the partial cell thickness. I suppose it would also be the right thickness to use for volumetric flow in horizontal directions through the cell. Perhaps the distinction between the "gridpoint" and "narrowed" cell dimension is not required for thickness? If it is needed, we may have to return to that one sometime!

> Users have to rely on the `long_name` or other attributes to know which point on a grid cell the thickness refers to. Could use cases for `cell_x_width` be specified in the `long_name` attribute? This might give some flexibility to how it is used.

Certainly you could, but that not be a CF convention, which doesn't standardise the contents of `long_name`.

Best wishes

Jonathan

**Comment by jmecki** [comment]
Hi All,

Sorry for not responding in a while.  As suggested by @atb299 I think the names cell_x_width and cell_y_width seem like a good choice.  It is a 2D time invariant variable which can be defined for different grid points (i.e. T,U and V points)

Also, to add to the discussion similar to cell_thickness how was cell_area accepted?

What still needs to be done to have a decision for a name for this variable or what information is still required?

Kind Regards,
Jenny

**Comment by atreguier** [comment]
Hi Jenny,

The problem is that in NEMO these quantities must be utilized only to compute transports, and thus are required at U and V points of the C grid. The width used to compute tracers (heat and salt content, for example) are not the same and are not modified in the straits, as shown by the pages of the NEMO manual I added to my nov 9 post. I don't know how it is done in NorESM, but NEMO is used by many groups in CMIP and thus the names should at least be clear for NEMO users. 

I would favor rather something like cell_x_width_velocity_scale_factor and cell_y_width_velocity_scale_factor, a bit like @atb299 but with "velocity" added. this would give the hint to the user that they are applicable to velocities. 

Anyway, the best if for modellers to output directly umo and vmo (the mass transports) on the native grid. These are among the baseline variables, I think. But unfortunately your found that the groups didn't all do it for CMIP6.

Anne Marie

**Comment by jmecki** [comment]
Thanks Anne Marie,

I'm still unclear why it is discouraged to include cell_x_width and cell_y_width in a similar way to how areacello is included in the CMIP output.  It would be very useful to have them output for all the grid points (i.e. T, U, V points), regardless of whether or not the model uses narrowed straits.  They would also be useful on T-points not just U and V points since it would make computing section averages much easier.  Also, in CMIP models areacello is only available on T-points.

However, I think that whether or not these variables are included or should be included in CMIP output is another issue.  And as a first step would be to have a standard name for them, whether or not it's output is another issue.  It would be great to have one in a similar way areacello has a name.

Jenny

**Comment by jmecki** [comment]
Hi Jonathan,

The quantities I was thinking of are lengths not scale factors.  The example of the narrow channels are a region where having these variables would be extremely useful.  But I think that a 2D fields containing dx (cell_x_width), and dy (cell_y_width) for every grid box and that can be defined for each grid point (i.e. T,U,V) would be useful none the less.  The relationship with areacello should be dx*dy = areacello.   I hope that makes more sense.  I agree that length is clearer than width.

Jenny

**Comment by JonathanGregory** [comment]
Shall we add _four_ new standard names, then:
```
cell_x_length
cell_y_length
cell_x_length_for_velocity
cell_y_length_for_velocity
```
2D fields of these quantities would be fine in CF. What should be archived in CMIP isn't a CF question, of course.

**Comment by JonathanGregory** [comment]
Yes. CF doesn't have a way to describe Arakawa grid relationships, though it's been mentioned a few times. However, if you provide these cell-length fields as 2D, the user can compute some things, just by using the metric on the appropriate grid, without having to understand Arakawa grids.

**Comment by jmecki** [comment]
>  However, if you provide these cell-length fields as 2D, the user can compute some things, just by using the metric on the appropriate grid, without having to understand Arakawa grids.

I agree, making having these variables very handy :)

**Comment by japamment** [comment]
Dear Jenny, @jmecki 

Thank you for your proposals in this issue and for pursuing the discussion.

Thanks also to those who have contributed helpful questions and comments to the discussion @ChrisBarker-NOAA @taylor13 @atreguier @atb299 @martinjuckes @larsbarring and in particular to Jonathan @JonathanGregory for coming up with a formulation for the names that it seems can be agreed.

Since the names themselves seem close to agreement I have summarized them again below and added description text based on the discussion here and existing names.

`cell_x_length`
**Units:** m
**Description:** "Cell" refers to a model grid-cell. "Cell_x_length" is the length of the grid-cell in the x direction of the model grid.

`cell_y_length`
**Units:** m
**Description:** "Cell" refers to a model grid-cell. "Cell_y_length" is the length of the grid-cell in the y direction of the model grid.

`cell_x_length_for_velocity`
**Units:** m
**Description:** "Cell" refers to a model grid-cell. The quantity with standard name cell_x_length_for_velocity is the length of the grid-cell in the x direction of the model grid used to calculate flow in channels that are too narrow to be accurately represented at the horizontal resolution of the grid. The cell_x_length_for_velocity is less than the length that can be calculated directly from the grid coordinates, which has the standard name cell_x_length.

`cell_y_length_for_velocity`
**Units:** m
**Description:** "Cell" refers to a model grid-cell. The quantity with standard name cell_y_length_for_velocity is the length of the grid-cell in the y direction of the model grid used to calculate flow in channels that are too narrow to be accurately represented at the horizontal resolution of the grid. The cell_y_length_for_velocity is less than the length that can be calculated directly from the grid coordinates, which has the standard name cell_y_length.

@jmecki are you okay with these? If so I think they can be accepted for publication now.

Best wishes,
Alison

**Comment by ChrisBarker-NOAA** [comment]
I'm a bit wary -- have we heard from folks from more that one modeling community?

IIUC, the folks asking for this are from the NEMO, and a similar concept was found in I NorESM. But it's not clear to me that they use them in the same way-- I haven't dug deep, but at the surface, it seems that the  NorESM uses this to adjust the model-predicted (and reported?) velocity to get a "true" velocity for determining mixing, etc. which means if you want, e.g. flux, you would use the reported velocity and the standard cell size.

But with NEMO, you could use the reported velocity and this "modified" cell size.

Which *may* be OK, I suppose the name is the still the same, it's actually the reported velocity that has a different meaning.

Do any other models use this concept? I haven't seen it in ROMS, but I've only looked at the output -- maybe it's used internally, and not reported.

Note: what I see in ROMS output is: 

```
standard_name: sea_water_y_velocity
long_name: v-momentum component
units: meter second-1
time: ocean_time
cell_methods: ocean_time: point
```
Interesting that it's a velocity (which is how I use it) but the long name is "momentum" -- related, yes, but ???

Anyway, the point is that you need to understand the model to know how to use these parameters anyway :-(.

But if folks think that these terms really does mean the same thing in all (most? some? more than one?) the models that use them, then a standard name makes sense.

## NOTE 2:

Perhaps the ship has sailed on this, but I'm uncomfortable with the concept of "grid cell" dimensions -- it makes perfect sense for rectangular grids, but for non rectangular grids (even logically rectangular) not so much:

A 3D (quad) cell is defined by six faces (and 12 edges) -- and the quantity that we're talking about is not the "cell width" but actually the "face width" -- and could (will?) be different for two faces of the same cell.

In fact, in this discussion, we've talked about defining this on the "U points" and "V points", but those are really the U and V faces (or edges, depending on if you are thinking 2D or 3D.

Anyway -- at this point in CF, I don't think the whole edge-face-cell thing is defined (outside of UGRID) -- so maybe stay the course, but we've been warned :-)

**Comment by jmecki** [comment]
@japamment I am happy with those names, with the comments from @ChrisBarker-NOAA it might be worth noting in the description that they are relevant for rectangular grids.  Perhaps slightly changing the wording to 'rectangular model grid' or 'quadrilateral model grid' as opposed to just 'model grid'.

**Comment by jmecki** [comment]
Hi All,

There has been some offline discussion on this topic between @japamment , @atreguier and I.  Below is also sketch to help explain some of the details of the variables being asked for:

![Image](https://github.com/user-attachments/assets/ae36543d-6f4f-4338-8d73-1a0d4fe317b9)

Through our offline discussion we have come to the conclusion that the most logical thing would be to have the following names (reference to the sketch above to make it easier to follow):

dxto = cell_x_length
dyto = cell_y_length

dxuo = cell_x_length_for_x_velocity
dyuo = cell_y_length_for_x_velocity

dxvo = cell_x_length_for_y_velocity
dyvo = cell_y_length_for_y_velocity

As pointed out in the previous discussions the full set of these variables are more relevant for C-grid models, with dxto and dyto being most important for B-grid models where it's easier to estimate what is needed for the more detailed computations.  Also, in B-grid models dxuo=dxvo, dyvo=dyuo, which is not the case for C-grid models.  Having names for these variables means they can be formally defined when they are output/used.  However, having names for these variables should not influence whether or not they are included in the list of variables output as part of the CMIP dataset or other any other model study output, that is a discussion that should be happening outside of this one.

The point about the artificial narrowing of some channels has been mentioned several times (see previous discussions for more details).   This is something that is often done for the NEMO ocean  model in courser resolution simulations like the ones use in CMIP6.  In my opinion, this is only a part of the reason to have names for these variables and they are useful regardless of whether or not the model uses artificial narrowing of channels.

**Comment by davidhassell** [comment]
Hi Jenny - Thank you for the nice picture!

I have a question, that I only thought of having seen the diagram ... A quick search on "units" in this issue shows that the canonical units of these names to be "m". Grids are often defined in (some sort of) "degrees", though. In this case the X length of a grid cell in metres will typically vary at different Y locations through the grid cell - more noticeably at higher latitudes. Is this a problem? Is the value representative of the whole grid cell in some way? If so, then this should perhaps be in the [descriptions](https://github.com/cf-convention/vocabularies/issues/219#issuecomment-2592060663). Calling it a "cell length" without qualifying what that means in the name seems not so appropriate to me, now. Words like "effective" were being suggested earlier, but I couldn't tell on a quick re-read why they were not considered for the current names.

I am a bit reluctant to restart things which are all but settled, so am happy to leave this if no one picks up on it :)

David

**Comment by jmecki** [comment]
@davidhassell having the units be in 'm' means they would be similar to the thkcello and areacello variables.

**Comment by davidhassell** [comment]
Hi Jenny,

OK .., but thickness and area are well defined in units of "m" and "m^2" respectively, meaning that there is only one value and it is unambiguous as to what it represents. The length doesn't enjoy that simplicity, I think. 

I'm not suggesting that "m" is an inappropriate unit, just that it could be a bit clearer to users exactly what the value in metres is representing.

**Comment by JonathanGregory** [comment]
Dear Jenny and David

I think the names Jenny has proposed are fine. The word `length` is appropriate for a quantity in `m`, and appears in other standard names. David is right that the grid itself is usually described by latitude and longitude, but these grid metrics will be easier to use if they are given as lengths, so they can be multiplied by velocities. In their descriptions, I think we should advise using the names `for` velocity only when the more general `cell_x_length` or `cell_y_length` is not applicable. I wonder why you need the lengths parallel to the velocities as well as perpendicular - maybe it's for calculating derivatives?

Best wishes

Jonathan

**Comment by atb299** [comment]
Hi Jonathan,

> I wonder why you need the lengths parallel to the velocities as well as perpendicular - maybe it's for calculating derivatives?

As far as I'm aware, you don't - I've seen/heard of channels being artificially narrowed, but never shortened. But the names need to be flexible enough to allow a cell length to be provided for restriction in either the x or y direction.

**Comment by ChrisBarker-NOAA** [comment]
Sorry to throw too much sand in the works this late in the game, but:

1) Is it "velocity" that's the issue, or "flux"? I think flux may be the better term (closely related as well).

2) 
> In this case the X length of a grid cell in meters will typically vary at different Y locations through the grid cell

This has bothered me as well -- using "length" of a cell only really makes sense for rectangular cells -- and aside from the distortion that comes from the round earth, there are also logically rectangular quad cells that don't have a clearly defined "length". I *think* NEMO falls into this category, and certainly ROMS does.

ROMS (A C-grid model), at least, puts the "velocities" on the cell edges -- so what you want is the length of that edge, not the length of the cell itself.

But I don't think that CF yes has the concept of "edges" -- at least outside of UGRID.

**Comment by ChrisBarker-NOAA** [comment]
> NEMO (also a C-grid model) provides cell dimensions centered on each point on the grid. There is a cell length, width and thickness for the T, U, V, F and W points. Velocities are on the cell edges, and as you say, it is the lengths of these edges that are required.

which is what confused me -- e.g. there are two U values, on the opposite edges of the cell. Those edges are not necessarily the same length. Is it correct to use the (presumably average) value for the "width" of the overall cell? In practice, cells aren't all that distorted, so it probably doesn't matter, but we should be clear about what a "cell length" really means.

**Comment by JonathanGregory** [comment]
@jmecki, is there a present use-case for `cell_x_length_for_x_velocity` and `cell_y_length_for_y_velocity`? If there is, it would be interesting to know the application, and probably relevant to mention it in their description. If not, maybe we don't need to define those standard names just now. Since we'd have established a pattern, it would be easy to add them if required in future.

@ChrisBarker-NOAA, I suppose that "a cell length for velocity" means a grid metric which is used by a model to do certain calculations e.g. compute fluxes from velocities. It's defined by that use, and it's calculated however would be consistent with the model concerned. It's a model concept, but we have plenty of other such standard names e.g. ones which mention `model` or refer to concepts like "diffusivity".

**Comment by atreguier** [comment]
Dear all, what is finally the list of variables that would be created? 

Just to verify:
cell_x_length_for_temperature  , corresponding to proposed CMIP7 variable: dxto
cell_y_length_for_temperature  : dyto
cell_x_length_for_x_velocity     :  dxuo
cell_x_length_for_y_velocity     :  dxvo
cell_y_length_for_x_velocity.    :  dyuo
cell_y_length_for_y_velocity.    :  dyvo

I am afraid the "temperature" points may have been forgotten in the latest discussions.

Anne Marie

**Comment by jmecki** [comment]
@atreguier would it be better to use for_tracers instead of for_temperature?  Otherwise it might get a bit confusing.

@JonathanGregory these values would be useful to define for consistency.  Also, they can be used to compute derivatives (e.g. dT/dx can use dxu on a C-grid)

**Comment by JonathanGregory** [comment]
Dear Anne Marie and Jenny

Standard names imply that the quantity named is for general purposes if there _isn't_ a qualifier of some sort, meaning the quantity includes everything, or applies under all circumstances, etc. If quantities named as `cell_x_length` and `cell_y_length` were provided, and no other cell sizes, you would use those ones for all purposes. My understanding is that the proposal comes from the particular need for a "modified" length for flux calculations for some gridboxes. Therefore I think we need only to provide standard names for cell lengths, needed for particular purposes, which differ from the "usual" ones.

I think the quantity `cell_x_length_for_y_velocity` should be defined or described in some terms which say what it means, or what its purpose is; for instance, it is the cell width needed for calculations concerned with y-velocity. It shouldn't be defined as the x-size of velocity gridboxes, for instance, although that's probably how it is obtained, because that is specific to the construction of the grid. We do not need standard names to distinguish cell metrics for tracer and velocity grids for general purposes. Both grids can use `cell_x_width` and `cell_y_width`, because their coordinates and dimensions will distinguish them.

Is the `dxu` you need for a derivative on the C-grid the "ordinary" velocity cell x-width, or is it the modified one for narrow channels, which motivated the need for special names?

Best wishes

Jonathan

**Comment by atb299** [comment]
> > NEMO (also a C-grid model) provides cell dimensions centered on each point on the grid. There is a cell length, width and thickness for the T, U, V, F and W points. Velocities are on the cell edges, and as you say, it is the lengths of these edges that are required.
> 
> which is what confused me -- e.g. there are two U values, on the opposite edges of the cell. Those edges are not necessarily the same length. Is it correct to use the (presumably average) value for the "width" of the overall cell? In practice, cells aren't all that distorted, so it probably doesn't matter, but we should be clear about what a "cell length" really means.

Hi @ChrisBarker-NOAA , its a grid, so the two U values have different indices and, as you say, the cell lengths that correspond to each of these points are not necessarily the same. The For NEMO T(i,j) is equidistant between U(i,j) and U(i-1,j):

![Image](https://github.com/user-attachments/assets/47fe8706-e3db-40fa-9e90-89baef3d0fb2)

**Comment by atreguier** [comment]
Hi, @JonathanGregory , I did not follow the beginning of the discussion, so I'm not sure how I can help... if more description is needed, where must these descriptions/definitions be submitted? 
@jmecki , I agree that  cell_x_length_for_tracers is more appealing to the Nemo community (we talk about the "T" points as "tracer" points) but I am afraid this may lead to discussions like "what is a tracer? which tracers? which units"? etc etc. If there is no danger of this happening, then yes, I prefer "tracers" as well.

**Comment by jmecki** [comment]
@japamment in the description can tracers and/or examples of tracers be added to the description, regardless of if for_tracers or for_temperature is used?

@atreguier I worry the other way around, that people might not understand that for_temperature should also be used for salinity...

**Comment by taylor13** [comment]
It appears that many of the names propose apply only to specific types of grids, so should the "grid type" be included in the standard name?    There may be a problem also defining these in terms of different associated variables (e.g., tracers vs. velocities).  It's true that velocities may be calculated on cell interfaces, but values are often reported (for some uses) at the same location as the tracers.  So cell_x_length_for_x_velocity would depend on *which* on which of the two grids were used.

I wonder if the "ugrid"conventions might be a better way to describe more completely the grid structure. 

Perhaps for now, we should ***not*** try to address the general "grid description" problem and deal with the seemingly special case (see top of this issue) of how to indicate the modified width of some grid cells (in certain models), which is done in an attempt to better represent processes in narrow straights (e.g. vertical mixing that depends on vertical shear in horizontal velocity).

**Comment by atreguier** [comment]
Hi, 

I like to compute quantities in a way consistent with the model code, it is necessary when you want to close budgets. In every model on a C grid, the grid lengths are used in the discretized equations. cell_x_length_for_y_velocity is called "e1v" in NEMO,  "dxCv" in Mom6, etc.  I have not seen any way to submit those variables on esgf. The grid cells are not exact rectangles. An approximate computation of these lengths (as opposed to taking the ones that were actually used by the model) sometimes results in sizeable errors when, e.g., you re-compute a meridional overturning of a transport across many cells. This happens even if there has not been any local modification of the cell lengths.

If there was a way to submit these grid variables to esgf, I suppose that they would be found useful for ocean analyses of all the models that use rectangular grids. Jenny's proposal of the new variables dxu, dxt, etc has been welcomed positively by the "ocean and sea-ice" CMIP7 data request group. 

I am not sure that I understand well the relation between cf names and cmor names. Could there be less cf names than Cmor names?  for example: a generic "cell_x_length_for rectangular_grid" that would be used for variables Ofx.dxu, dxv, dxt, and a generic "cell_y_length_for_rectangular_grid" that would be used for variables Ofx.dyu, dyv, dyt?

**Comment by JonathanGregory** [comment]
Dear Anne Marie, Jenny, Karl, Adam et al.

The purpose of standard names is to distinguish and identify quantities in scientific terms. The description of grids isn't part of the function of standard names; that's done by the coordinates. Therefore I don't think the standard names should include any information referring to Arakawa grids or describing the grid type. For the same kind of reason, for instance, `air_temperature` always has the same standard name regardless of whether it's on height levels, pressure levels or any other kind of vertical coordinate surface.

This relates to Anne Marie's last question. There is no relationship between CF standard names and CMOR names. Exactly as you say, there are fewer CF standard names than CMOR names. CMOR names correspond to distinct data requests, and involve the choice of grid as well as the scientific quantity in their definition.

I agree that it's important to be able to compute quantities in a way which is consistent with model code. That is why CF provides ways to store areas and volumes of cells, and that's why you want to store and identify the relevant grid lengths. For this purpose, it makes sense to to provide data variables with standard names `cell_x_length` and `cell_y_length`, both (_y_,_x_) where _x_ and _y_ are the horizontal dimensions, to contain the values actually used by the model for each gridcell to calculate derivatives, fluxes and so on. If there is more than one grid, there will be different cell-length data variables for each grid (e.g. the four grids shown in Adam's diagram).

It is fine to have four variables with the standard name `cell_x_length`, although they can't have the same netCDF variable name of course if they're in the same file. They can be distinguished by the dimensions or coordinates. That's a bit laborious, however. If it's useful, we could add a new feature to the CF convention, so that the appropriate metric variables could be linked to each data variable using the `cell_measures` attribute. We do this for cell area and volume, for the same reason of convenience. It would look like something like this on an Arakawa B-grid:

```
float opottemp(level,yt,xt);
  opottemp:standard_name="sea_water_potential_temperature";
  opottemp:units="degC";
  opottemp:cell_measures="area: areacellot x_length: dxt y_length: dyt";
float uo(level,yu,xu);
  uo:standard_name="sea_water_x_velocity";
  uo:units="m s-1";
  uo:cell_measures="area: areacellou x_length: dxu y_length: dyu";
float dxt(yt,xt);
  dxt:standard_name="cell_x_length";
  dxt:units="m";
  dxt:long_name="distance across tracer cells in the x direction";
float dyt(yt,xt);
  dyt:standard_name="cell_y_length";
  dyt:units="m";
  dyt:long_name="distance across tracer cells in the y direction";
float areacellot(yt,xt);
  areacellot:standard_name="cell_area";
  areacellot:units="m2";
  areacellot:long_name="area of tracer cells";
float dxu(yu,xu);
  dxu:standard_name="cell_x_length";
  dxu:units="m";
  dxu:long_name="distance across velocity cells in the x direction";
float dyu(yu,xu);
  dyu:standard_name="cell_y_length";
  dyu:units="m";
  dyu:long_name="distance across velocity cells in the y direction";
float areacellou(yu,xu);
  areacellou:standard_name="cell_area";
  areacellou:units="m2";
  areacellou:long_name="area of velocity cells";
```

For a lot of this discussion we were talking about "modified" cell lengths, say `dxu_strait` and `dyu_strait`. I understood that these are like `dxu` and `dyu` (computed as differences between tracer gridpoints), except that they take narrow channels into account, so that at some gridcells they are smaller than `dxu` and `dyu`. Computing the flux through the face of a cell in the x-direction involves the product of `x_velocity` and `dyu_strait`, for instance.

The proposed standard names `cell_x_length_for_y_velocity` and `cell_y_length_for_x_velocity` would be appropriate for `dxu_strait` and `dyu_strait`. They're different quantities from the usual `cell_x_length` and `cell_y_length`, because they aren't gridpoint separations, they're values used for computing fluxes. Is that right? The `for_`[`xy`]`_velocity` part of the name is there to indicate their scientific use, not to indicate which grid they're on. Included in the above example, they would look like this:


```
float dxu_strait(yu,xu);
  dxu_strait:standard_name="cell_x_length_for_y_velocity";
  dxu_strait:units="m";
  dxu_strait:long_name="distance across velocity cells for calculating fluxes in the x direction";
float dyu_strait(yu,xu);
  dyu_strait:standard_name="cell_y_length_for_x_velocity";
  dyu_strait:units="m";
  dyu_strait:long_name="distance across velocity cells for calculating fluxes in the y direction";
```

Does this make sense?

Happy weekend

Jonathan

**Comment by taylor13** [comment]
Earlier today (before Jonathan's post), I started to compose a comment about this but got interrupted.  My comment started:

> Apologies if I've missed something, but if this issue is about defining characteristics of cells and not simply recording the cell widths that are modified for certain purposes, then I think we should consider expanding "cell_measures" to include an option like "effective_cross_section".  Just as the cell_measures can record the area of a cell, we could have it record the "width of the cell orthogonal to the direction of the dimension.

So I think that two of us at least support exploring the use of cell_methods to record this sort of information.  Like "cell area" the cell width is useful for those analyzing data.
Karl

**Comment by atreguier** [comment]
Hi @JonathanGregory , 

The solution to have "cell_x_lenth" and "cell_y_length" as standard names seems fine to me. I don't think we need "modified" cell lengths, because once the cells are are modified before running the model, the original values are never used. dyu would be the y-width of "u" cells, used to calculate zonal transport everywhere, whatever the way it has been computed or modified. 

The issue of modified lengths is one of the motivations to publish e.g., dyu on esgf, because in that "modified" case if you try to re-compute the dyu approximately from the coordinates of neighbouring points you will not be able to recover something that could approximate  the "true" value used by the model at that specific location. 

I have never used the "cell_measure" attribute myself, but your suggestion seems great.

**Comment by JonathanGregory** [comment]
@atreguier, wouldn't you need the unmodified cell lengths (= distances between points on another grid) to calculate derivatives consistently with the model code?

**Comment by atreguier** [comment]
Hi @JonathanGregory , here is the page in the NEMO manual that explains the narrowing of straits: https://github.com/user-attachments/files/17686527/nemo_manual_handmade_grid_corrections.pdf .
To my knowledge, this is done when the width of one single grid point is significantly larger than the width of the channel in the real world. So, only channels that are one grid point wide are narrowed, and there is land on each side, so the derivative of the along channel velocity in the cross channel direction is never used anywhere. The velocity derivatives that are computed using the modified scale factors are all masked because they fall in land.

**Comment by JonathanGregory** [comment]
Thanks, @atreguier. I see, yes, the unmodified cell length is not used in the model code. However, that manual page explains that the cell area is still the product of the _unmodified_ cell lengths and hence the cell volume depends on the unmodified length too. For that reason I think we ought to stick to what we'd discussed before: define the plain `cell_x_length` and `cell_y_length` standard names for the lengths that relate only to the geometry of the grid, and also define `cell_x_length_for_y_velocity` and `cell_y_length_for_x_velocity` for the modified ones used for flux calculations. Is that OK for you and Jenny?

As Karl @taylor13 and I suggested, we can also define new `cell_measures` to make it easy to find the right variables for the grid of interest. That would be a conventions enhancement, not a standard name issue.

**Comment by atreguier** [comment]
@JonathanGregory, you quote "the cell area is still the product of the unmodified cell lengths and hence the cell volume depends on the unmodified length too". However, the volume mentioned here is the volume of the tracer cell, dxt*dyt. The two lengths at T point, dxt and dyt are never modified. The manual says that only the lengths at u or v points can be modified without losing consistency (heat conservation, for example). The "unmodified" dyu and dxv are never used and should never be used when analyzing the model results. 

What you propose will be a bit confusing for NEMO users, because the information about unmodified dyu, dxv is not kept around in the files we use (meshmask, meshgrid...). This information no longer exists. For a nemo user, there is only one dyu array and one dxv array. Does your proposal imply that we would need to define different cmor names for modified/unmodified dyu and dxv? That would complicate things, I think. An automatic analysis program based on the cmor name "dyu" would fail when the width has been modified at some grid point, because in that case the variable "dyu" would not be published and replaced by "dyu_modified" or something like it. This would be a pity. Jenny is on vacation until march 17th, she can't answer now, but I think that her intent was to facilitate model analysis, not complicate it. 

The process of modifying a strait width by hand is the same thing has modifying the bathymetry to ensure that a strait is open, or deep enough in the model's world. All modelling groups modify the bathymetry, it is necessary when you run a 1° model. I don't think anybody has suggested that the model groups publish their original (unmodified) bathymetry on esgf. What is needed for the analysis is the bathymetry that the model has been run with. Same for the dyu, dxv: what we need on esgf is the variables that should be used to analyse the model. 

Does this analogy help make my case?

**Comment by JonathanGregory** [comment]
Dear Anne Marie @atreguier

Thanks for further explanation. I'm sure that no-one wants to complicate the work of analysis - certainly not me. I have been seeking to clarify whether there were any purposes for which the unmodified cell length would be needed, because earlier in this discussion it was not clear. If the modified cell length is used for _all_ purposes, both in analysis and the model code, and you can't imagine a physically useful reason for needing the unmodified grid spacing instead, then I agree with you that we only need `cell_x_length` and `cell_y_length` standard names for the NEMO data. The description should define them as being the extent of the cell in the relevant dimension according to its cell boundaries and the model code. It would probably be helpful to remark in the description that they are not necessarily equal to the spacing of points on another grid of the model.

Best wishes

Jonathan

**Comment by atreguier** [comment]
Thanks! 
We are all set then! 
Please let me know if I need to do something in the coming weeks, while Jenny is away.
Cheers, 
Anne Marie

**Comment by ChrisBarker-NOAA** [comment]
>> which is what confused me -- e.g. there are two U values, on the opposite edges of the cell. Those edges are not necessarily the same length. Is it correct to use the (presumably average) value for the "width" of the overall cell? In practice, cells aren't all that distorted, so it probably doesn't matter, but we should be clear about what a "cell length" really means.

> Hi @ChrisBarker-NOAA , its a grid, so the two U values have different indices and, as you say, the cell lengths that correspond to each of these points are not necessarily the same. The For NEMO T(i,j) is equidistant between U(i,j) and U(i-1,j):

And the cell_x_length and cell_y_length would be defined on the U and V points?

If so, then  I *think* that this is all OK, but I'm not completely thrilled with the terminology. The problem is that the U and V "points" are really on the edges of the cell (sometime treated as points, sometimes as the full edge (e.g. for flux).

So I think the quantity we are talking about is not the "*cell* length", but really the "*edge* length"
 -- but so far, in CF, we don't have any way to talk about cell edges -- just cells [*].

So the question is -- *if* CF adds the concept of edges some day (e.g. something like SGRID) -- would it still work for the standard names of cell_?_length to be used for this same quantity? 

I'm thinking probably yes, so we're OK.

[*] -- well, we do have edges in the UGRID spec -- and in that case, "cell length" makes no sense, but "edge length" does (though there's less need for that -- unstructured grids are designed to avoid these challenges :-)

**Comment by davidhassell** [comment]
Hello, 

I think that [the descriptions posted earlier](https://github.com/cf-convention/vocabularies/issues/219#issuecomment-2592060663), which provide a really useful reference, have been superseded by the subsequent discussion. Might it be possible for someone with all this in their head to post some new standard name descriptions?

Thanks,
David

**Comment by JonathanGregory** [comment]
Dear @atreguier et al.

Thanks for your helpful text. I have some suggestions:

* Although these are most likely to be used for model data, they could apply to any gridded dataset. Perhaps we could omit the first sentence? ("'Cell' refers to a model grid-cell.")

* Because "length" might not be self-explanatory, I suggest we use a different phrase to describe it in the second sentence. For instance, "`cell_x_length` is the distance across the grid-cell in the x-direction of the grid."

* Since CF doesn't associate the staggered grids, we don't have a notion of corresponding cells on different grids, so it's not quite correct to say that we'd define multiple lengths for "each cell".

* It occurs to me that the point of the final sentence doesn't only apply to staggered grids. It's possible that even on an A-grid you might have cell lengths which aren't equal to grid point separations. Hence I would suggest separating these two remarks:

>  Although the locations of the grid points may be geometrically related, the cell lengths are not in general required to be equal to the spacing of grid points. In the case of staggered grids (for instance, an Arakawa B-grid), different variables in a dataset may have different locations in space (e.g. tracers or velocities), and each grid has its own cell lengths.

Best wishes

Jonathan

**Comment by JonathanGregory** [comment]
Yes, I agree, that's a better way to say it, @davidhassell. Thanks.

**Comment by atreguier** [comment]
agreed as well, thanks.

**Comment by ChrisBarker-NOAA** [comment]
> Although the locations of the grid points may be geometrically related, the cell lengths are not in general required to be equal to the spacing of grid points. In the case of staggered grids (for instance, an Arakawa B-grid), different variables in a dataset may have different locations in space (e.g. tracers or velocities), and each grid has its own cell lengths.

Hmm -- there's only one grid -- "each grid" doesn't make sense here. 

maybe: "different variables in a dataset may be defined at different locations on the cells (e.g. tracers or velocities), and each location on the cell can have different cell lengths"

???

which is why I'm uncomfortable with "cell" here -- it may not be the length of a cell, but rather the length of a cell edge -- but no one else seems to share that discomfort, so here we are.

**Comment by taylor13** [comment]
I also think of one grid of cells with cell vertices and edges and areas and such.  But I haven't thought of another term (yet).

**Comment by atreguier** [comment]
Hi, 

The lengths may not all be easily defined as "edges". In rectangular grids the area may be defined by the product of two lengths that are not "edges". In NEMO, areacello (cell centered on a T point) =dxt*dyt (lengths centered at T point, in the x and y directions).

**Comment by davidhassell** [comment]
Some thoughts on "cells" ...

In the [CF data model](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.12/cf-conventions.html#data-model-coordinates), the data of _one_ variable are defined on the cells of a grid (i.e. the "domain"), and each cell describes the extent (typically spatio-temporal, but not necessarily) to which a value of that data applies. The data of any other variables are of no consequence here. A cell is $N$-dimensional ($N\ge 0$), and may even be non-contiguous (e.g. climatological cells, geometry cells). The edges or vertices of a cell with 1 or more dimension are not themselves cells - rather they are lower-level elements that define the cell's extent. This is not to say that those same vertices or edges can't be re-purposed to form the cells for a different variable's data.

(Aside: UGRID explicitly links up to three different grids into one "mesh" - those defined by nodes, the edges connecting those nodes (i.e. vertices), and the faces enclosed by those edges. This allows you to know the relationship between two variables defined at different mesh locations, but does not mean that data defined on any one of these grids is also defined on the other two.)

If a cell is 1-d (an "edge")  then what does `cell_x_length` mean - is it the length of the X component of the vector that you'd follow when traveling between the edge's two vertices? Or would a name of `cell_length` be more appropriate, defined as the distance along the edge?

**Comment by JonathanGregory** [comment]
Dear all

Putting together and elaborating the words which David and I previously suggested, for the definition we could say

> `cell_x_length` is the linear extent of the cell in the x-direction of the grid. For a rectangular cell whose sides are parallel to the Cartesian axes of a plane, the `cell_x_length` is the distance between the two sides in the x-direction. For the usual case of horizontal grids in geoscience, where the grid cells are not planar or rectangular, the `cell_x_length` is a distance which is representative of the extent of the cell in the x-direction. Generally it is not uniquely defined, but for a model dataset it should be a distance which is consisent with the model's own computation of extensive quantities e.g. fluxes or areas.

That's quite a lot, but is it clear? Is it what you have in mind, Anne Marie @atreguier?

Chris is right that a my reference to "grids" isn't satisfactory, since there's only one grid for a given data variable. For that part of text, how about

> Although the locations of the grid points may be geometrically related, the cell lengths are not in general required to be equal to the spacing of grid points. When a dataset contains variables on more than one grid (for instance, the Arakawa B tracer and velocity grids), each grid has its own cell lengths.

Best wishes

Jonathan

**Comment by atreguier** [comment]
Hello, 
I don't understand what you mean by "equivalent rectangular cell". We wish to have the exact lengths for the cells that a given model is using. 

Let's consider a typical ocean model, with the equations discretized on a curvilinear orthogonal grid on the sphere, and the variables staggered in space following what we call an Arakawa B or C grid (most ocean models have these characteristics: GFDL and ACCESS, NCAR, NEMO-based models in Europe, NorESM, all the Chinese models...). When making calculations using the model results, it is more accurate to use the exact lengths that were used in the model equations. The suggestion of having a cf name is to make possible for modelling groups to publish these exact lengths. Of course these lengths can be approximated from grid points positions, but the approximation is not good enough for some types of analyses.

**Comment by ChrisBarker-NOAA** [comment]
>  When making calculations using the model results, it is more accurate to use the exact lengths that were used in the model equations. 

of course -- but what are they???

In a logically rectangular, curvilinear grid, the cells are non-rectangular quads -- there are four vertices, and four edges -- each edge has a length, the cell has an area, bu there is no one "cell length" or "cell width".

So what is used in the calculations??? I have no idea if all the, e.g. Arakawa C grid models do the same thing, but If you want to conserve mass, you need to know the flux though the cell edges, yes? so each edge is going to have a length (or area in 3D) -- if that's what you want, shouldn't we call it something like the edge length, not cell length?

What I've seen from ROMS output is that "tracers" are computed/defined on what ROMS calls the "rho points" -- which are at the center of the cells. What does it mean to have single "cell_x_length" defined on those points? It seems that is the length of an equivalent rectangular cell (that is can compute an area from it) -- but if you want area, maybe save the area?

On the other hand, on the "U points" or "V points" -- those are on the edges of the cells -- and to compute, e.g. flux, you'd need the length of that edge. IIUC, the proposal is to call that length the "cell_x_length" which seems potentially confusing to me. Also, a given edge is either the x direction or the y direction, so hopefully no one will define the x_length on a y-edge ....

The fundamental problem here is that CF has no way to fully define a curvilinear grid -- we can define the cells themselves (what ROMS calls the Psi grid) but not how the different pieces fit together (what some folks are calling different "grids" -- because CF has no way to define one grid with different pieces). It does talk about Cell connectivity, but I don't think that the concept of an edge as it's own thing is there.

If the geometry of the grid (e.g. the size of the cell) can be fully computed from it's nodal coordinates, then we're OK -- but this whole topic is about what to do when that's not the case.

Which is (one of ) the reason we introduced the UGRID standard -- yes, you can theoretically fully describe a UGRID as a set of independent cells with vertices, but it's a whole lot more efficient, and usefull, to describe the mesh as a mesh, including how the pieces fit together.

Again -- I'm not intending to block consensus here -- if most folks are happy with this, then fine, go ahead -- but as a user of these types of model results, this feels like a confusing and not-complete solution to me.

And ultimately, I'd like to see a complete solution, such as the SGRID proposal, introduced to CF.

(https://sgrid.github.io/sgrid/)

I don't think we should block this name proposal while we wait for that, but maybe keep it in mind when finalizing the names?

To the NEMO folks -- have you looked at SGRID? It would be great to know if that would work for NEMO as well and how -- there are currently examples from Delft3D, ROMS, and WRF.

**Comment by davidhassell** [comment]
Thanks, Chris. I agree with your concerns about non-rectangular cells. I'd be fine with restricting the description to rectangular cells, so as to avoid such ambiguities. (Although tripolar grids (like NEMO)  have rectangular cells, but they get very stretched and twisted in the northern high latitudes - again questioning "what is x", perhaps?)

> On the other hand, on the "U points" or "V points" -- those are on the edges of the cells 

I think this is can be a source of confusion. In many models, there are U and V _cells_ whose centres coincide with the mid-points of the edges of Rho cells (or at the vertices of rho cells, depending on the staggering).  We are talking about the "size" of _cells_, not the length of the common interface between adjacent cells. Right?

I need to remind myself about SGRID ...

**Comment by ChrisBarker-NOAA** [comment]
>> On the other hand, on the "U points" or "V points" -- those are on the edges of the cells

> I think this is can be a source of confusion. In many models, there are U and V cells whose centres coincide with the mid-points of the edges of Rho cells (or at the vertices of rho cells, depending on the staggering).

I'm thinking of C-grid models, where, yes, this is the case. But I finally figured out the (my) confusion.

IIUC:

In CF, a "cell" can be 1D, or 2D, or 3D.

This is why we used the terms "face" and "volume", rather than "cell" in UGRID (and SGRID)

 -- and a "grid" or "mesh" really isn't a concept at all, it's just a bunch of cells, for which you have to assume or compute the connectivity. (thought there are some features that make that easy for the Usual cases.

So for a curvilinear, quadrilateral, logically rectangular model grid, you have quadrilateral 2D cells, defined by the "psi points" -- those are the faces, in SGRID terms.  
(Link to the [classic ROMS grid diagram](https://www.myroms.org/wiki/File:Whole_grid.png), for reference)
the "rho points" are at the center of those quads, and the U and V points are on the edges of those quads.

But in CF terms, there is no overall "grid" -- only cells, so in that case, the rho points are on the 2D quadrilateral cells defined by the psi points, and the U and V points are on 2D cells defined also by psi points.

Though that gets back to my point -- if we want to know the length of the edge that, e.g. the U points are on (to compute a flux, for instance) then that's the length of a 1D cell, in CF parlance, not the x_length or y_length of a 2D cell.

But: 
> (like NEMO) have rectangular cells

Whoa! I'm sorry, I have been confused all along -- I thought NEMO supported curvilinear grids -- but if indeed, it is only rectangular, then the whole cell_x_length and cell_y_length does make sense! Sorry for all the noise caused by my confusion.

One question though:

In the case of an artificially narrowed cell (which is what this is all about, yes) -- where exactly is the narrowing applied?

What I mean is that if a cell is narrowed, then the presumably the adjacent cell must be consistent -- so the edge between those two cells is what is narrowed, rather than the whole cell -- so we are back to specifying an edge length, rather than a cell length.

Or maybe I'm missing something key here.

**Comment by atreguier** [comment]
Hi, NEMO supports curvilinear orthogonal grids. I suppose that when Chris said "rectangular grid" he just meant a polygon with 4 sides, not really a rectangle. We live on a sphere anyway. 
Most ocean models use tripolar grids, these are orthogonal curvilinear grids on a sphere. 
I didn't know about the sgrid proposal. @atb299 , have you heard about it? Has this been discussed in the NEMO community? 
Regarding the question about narrowing the straits, here is the page in the NEMO manual that explains it: https://github.com/user-attachments/files/17686527/nemo_manual_handmade_grid_corrections.pdf .
As I mentioned earlier in the discussion, only one-grid-point-wide straits are narrowed, so that the adjacent cells are on land and the grid lengths that fall on land are never actually used (any result of a calculation on land gets masked).

**Comment by atb299** [comment]
@atreguier, no I hadn't heard about the sgrid proposal.

**Comment by JonathanGregory** [comment]
I'm repeating, somewhat simplified, the text I suggested [above](https://github.com/cf-convention/vocabularies/issues/219#issuecomment-2671969828) in case it wasn't noticed:

> `cell_x_length` is the linear extent of the cell in the x-direction of the grid. For a rectangular cell whose sides are parallel to the Cartesian axes of a plane, the `cell_x_length` is the distance between the two sides in the x-direction. For the usual case of horizontal grids in geoscience, where the grid cells are not planar or rectangular, the `cell_x_length` is a distance which is representative of the extent of the cell in the x-direction. For a model dataset it should be a distance which is consistent with the model's own computation of extensive quantities e.g. fluxes or areas. In some cases, the cell lengths are equal to the spacing of grid points, but this is not generally so. When a dataset contains variables on more than one grid (for instance, the Arakawa B tracer and velocity grids), each grid has its own cell lengths.

As we've discussed, we could point to `cell_x_length` and `cell_y_length` variables from `cell_measures` of the data variable, as we can with cell area. Maybe we could require that `cell_x_length` &times; `cell_y_length` = `cell_area`? 

The idea of these standard names is solely to provide correct dimensions, which can be used without knowing the geometry, for working out other quantities. This is not the same aim as providing a description of the construction of the grid and its relationship to other grids, like SGRID. That would be a different issue.

**Comment by ChrisBarker-NOAA** [comment]
> Hi, NEMO supports curvilinear orthogonal grids. I suppose that when Chris said "rectangular grid" he just meant a polygon with 4 sides, not really a rectangle.

No -- when I said rectangular, I meant rectangular - I think someone else proposed that this could be for only rectangular cells. But apparently not, as the use case in mind is for more genereal quads -- thanks for making that clear (probably not for the first time...)

> We live on a sphere anyway.

Well, yes, but it's pretty common to model the earth as at least locally flat, so we can treat a rectangle as a rectangle in some instances.

> Regarding the question about narrowing the straits, here is the page in the NEMO manual that explains it:

Thanks for that -- here's a bit of it:

>  The key issue here is only to reduce the faces of T-cell (i.e. change the value of the horizontal scale factors at u- or v- point) but not the volume of the T-cell

Sorry, I'm not familiar with NEMO terminology, so not entirely sure what a T-cell is, but if I interpret that correctly, then it's as I suspected:

So (using 2D terminology):

The U and V points are on the edges of a quad cell -- so what you are adjusting is not the length of a quad cell, but the length of an *edge* of the quad cell (or the width of a face). and presumably, the length of that edge must match the length of the edge of the adjoining cell. Which means that some cells will have different lengths for, e.g. the U point on one end of the cell and the U-point at the opposite end. So what is the "cell length"??

and: "reduce the faces of the T-cell, not the volume".

Which I interpret to mean that you don't want to use the adjusted length to compute the area(2D) or volume (3D) the cell, so it's NOT to be interpreted as the "length of the cell" in other contexts.

What may be tripping me up is that in CF, a "cell" can be 1D or 2D, so the edges of those quads can be described as cells as well, in which case it is a "cell length" -- though I don't think you'd need the "x" and "y" in that case -- if it's on a "U point" then it's the x-length, if it's on the V point, it's a y-length.

**Comment by atreguier** [comment]
Hi @ChrisBarker-NOAA , I am sorry that I seem to have complicated the discussion rather than simplified it. 

Could we agree on @JonathanGregory' proposal?

**Comment by davidhassell** [comment]
> The U and V points are on the edges of a quad cell -- so what you are adjusting is not the length of a quad cell, but the length of an edge of the quad cell (or the width of a face)

I don't think so, but I might well have misunderstood :) In the use case here there are no U and V _points_, but there are U and V _2-d cells_ (whose centres happen to coincide with the edges/corners of the temperature grid cells). The best picture I could find on-line to show this is:

![Image](https://github.com/user-attachments/assets/616965f5-1b57-4864-b313-2cfb79099c31)

where the dark grey cell is for V, the (partially obscured) light grey cell is for U, and the white cells for mass/temperature.

**Comment by atreguier** [comment]
Hello,

Do you think we can wrap this up? I don't know how these github discussions come to a conclusion. 

cell_x_length
Units: m
Description: The linear extent of the cell in the x-direction of the grid. For a rectangular cell whose sides are parallel to the Cartesian axes of a plane, the cell_x_length is the distance between the two sides in the x-direction. For the usual case of horizontal grids in geoscience, where the grid cells are not planar or rectangular, the cell_x_length is a distance which is representative of the extent of the cell in the x-direction. For a model dataset it should be a distance which is consistent with the model's own computation of extensive quantities e.g. fluxes or areas. In some cases, the cell lengths are equal to the spacing of grid points, but this is not generally so. When a dataset contains variables on more than one grid (for instance, the Arakawa B tracer and velocity grids), each grid has its own cell lengths.

cell_y_length
Units: m
Description: Same as above, with "x" replaced by "y".

Is this OK?

**Comment by atreguier** [comment]
@davidhassell, I am OK with these changes.

**Comment by JonathanGregory** [comment]
David's version is fine for me as well.

Anne Marie, you asked how these discussions come to a conclusion. The answer is, when consensus is achieved, meaning there aren't any outstanding objections. It's not necessary for everyone to think that the result is best possible choice, but that everyone who's interested is willing to accept the result. Quite often discussions get stuck, and they require nudging to get them going again. It's useful to do that!

**Comment by ChrisBarker-NOAA** [comment]
@ taylor13: There is the follow on...

"For a model dataset it should be a distance which is consistent with the model's own computation of extensive quantities."

Which makes it clear for model results -- maybe that can be phrased to be a bit more generic?

Can anyone think of a non-model grid where the lengths won't match the geometric lengths? if so, that might inspire wording ...

**Comment by taylor13** [comment]
Sorry, I clearly missed that.  It's still a bit vague in that "consistent" with area isn't explicit, at least in the case of area  .. I guess the x length and y length when multiplied should give you the cell area.  

I'm happy to proceed with what we've got and not worry that we've focused on models in that sentence.  I suspect for observations, no one will report these lengths.

**Comment by atreguier** [comment]
The cell height exists already, the cf name is "cell_thickness".

**Comment by JonathanGregory** [comment]
As @atreguier says,

> The cell height exists already, the cf name is "cell_thickness".

This is because in standard names we use the word `thickness` for the vertical extent of a layer, in order to avoid confusion with `height`, which is the standard name for vertical distance above the surface.

**Comment by atreguier** [comment]
Dear @JonathanGregory , @ChrisBarker-NOAA 

@davidhassell proposed the following. 

cell_x_length
Units: m
Description: The linear extent of the cell in the x-direction of the horizontal grid. For a rectangular cell whose sides are parallel to the Cartesian axes of a plane, the cell_x_length is the distance between the two sides in the x-direction. For the usual case of horizontal grids in geoscience, where the grid cells are not planar or rectangular, the cell_x_length is a distance which is representative of the extent of the cell in the x-direction. For a model dataset it should be a distance which is consistent with the model's own computation of extensive quantities e.g. fluxes or areas. In some cases, the cell lengths are equal to the spacing of grid points, but this is not generally so.

cell_y_length
Units: m
Description: Same as above, with "x" replaced by "y".

Does everybody agree? We need the name for the CMIP7 data request tables. Will I get the information, when the cf name is accepted?

**Comment by JonathanGregory** [comment]
That's fine for me. The standard name moderators (@japamment @feggleton @efisher008) are responsible for the process. I don't know how it feeds into CMIP7 but I expect they can explain.

**Comment by davidhassell** [comment]
Fine for me, too!

**Comment by atreguier** [comment]
Hi @japamment @feggleton @efisher008 : can this discussion be closed? Please let me know if there is something that must be done!

**Comment by jmecki** [comment]
Sorry I was away for a while, but I'm happy with the latest suggestion by @davidhassell

**Comment by efisher008** [comment]
The names discussed in this issue were published in v91 of the Standard Name Table, so I am closing this issue as completed.
